Assets/Scripts/NPCs/StateMachine/States/Chase.cs

```diff
@@ -54,11 +65,42 @@ private Vector3 ComputeChaseVelocity()
 
         if (Vector3.Magnitude(velocity) > _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseMaxVelocity))
         {
-            velocity = (velocity / Vector3.Magnitude(velocity)) * _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseMaxVelocity);
+            velocity = (velocity / Vector3.Magnitude(velocity)) *
+                       _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseMaxVelocity);
         }
 
         velocity.y = 0;
 
         return velocity;
     }
-}
+
+    private void ComputeAStar()
+    {
+        Vector3 nodeWithoutY = Vector3.zero;
+        if (_currentTargetLocation != NPCsGlobalVariables.DefaultInitialVector)
+        {
+            nodeWithoutY = new Vector3(_currentTargetLocation.x, 0, _currentTargetLocation.z);
+        }
+
+        if (_currentTargetLocation == NPCsGlobalVariables.DefaultInitialVector ||
+            Vector3.Distance(_npcManager.transform.position, nodeWithoutY) <
+            NPCsGlobalVariables.WithinTargetNodeRange) // TODO Add check if we can see target node
```

Comment:
If we didn't reached the node yet BUT we can't see a straight line to it, then we should enter this if to recompute a new A* path

Reply:
You don't uderstand the issue here
If the NPC can't see the currentTargetLocation, it means that the node it is trying to reach is blocked by some sort of obstacle

I had the issue when a NPC was trying to reach a node near after a building corner, and was just constantly running in the wall and couldn't reach it

Reply:
This will be solved by collision avoidance in the #25

Reply:
okay